python/cudf/dataframe/string.py

```diff
@@ -558,13 +560,44 @@ def fillna(self, fill_value, inplace=False):
         result = result.replace(mask=None)
         return self._mimic_inplace(result, inplace)
 
+    def _find_first_and_last(self, value):
+        found_indices = self.str().contains(f"^{value}$").data.mem
+        found_indices = cudautils.astype(found_indices, "int32")
+        first = columnops.as_column(found_indices).find_first_value(1)
+        last = columnops.as_column(found_indices).find_last_value(1)
+        return first, last
+
+    def find_first_value(self, value):
+        return self._find_first_and_last(value)[0]
+
+    def find_last_value(self, value):
+        return self._find_first_and_last(value)[1]
+
     def unique(self, method='sort'):
         """
         Get unique strings in the data
         """
         result = StringColumn(self.nvcategory.keys())
         return result
 
+    def normalize_binop_value(self, other):
+        if isinstance(other, column.Column):
+            return other.astype(self.dtype)
+        elif hasattr(other, '__str__'):
```

Comment:
Since integers/floats have the __str__ attribute, I believe we would end up with something like this returning True:

```python
s = cudf.Series(['0', '1', '2'])
s == 2
False
False
True
```

This is unexpected behavior for this coming from pandas, in which all of these would be False.

```python
s = pd.Series(['0', '1', '2'])
s == 2
0    False
1    False
2    False
dtype: bool
```

Reply:
Ah thanks. So then should we just check if other is a `str`? Does that cover the common use cases? The only other value I can think of handling would be `None`

Reply:
That would likely cover the common use cases. I believe that users generally expect to be able to compare a string column to

- A single string (e.g., `'word'` or `'two words'`)
- Null representation (e.g., `None`, or `np.NaN` in pandas)
- A comparison array of the same length as the original array (in which case the broadcasting is unnecessary), perhaps such as a `column` (to use your example) but almost certainly also a `Series`.

Our normalization process should be designed so it can handle users supplying at least these three core types of comparisons.

Note that the behavior of comparing to None/np.NaN in pandas is interesting.

```python
s = pd.Series(['a', 'b', np.nan, 'c', None])
print(s)
print(s == None)
print(s == np.nan)
0       a
1       b
2     NaN
3       c
4    None
dtype: object
0    False
1    False
2    False
3    False
4    False
dtype: bool
0    False
1    False
2    False
3    False
4    False
dtype: bool
```

Finally, we should think carefully about what kinds of arrays we want to allow users to use as the comparison array. As of now, passing a `column` as the comparison array with Series backed by NumericalColumns will fail. We may want to be consistent with that behavior for StringColumns, or update it across the board to allow directly comparing against `columns`. Additionally, whether or not we intend to allow users to pass a CPU array such as `['a', 'b', 'c']` is a broader discussion that I think we can push down the road. @kkraus14's opinion on all of this would be valuable as well.

Reply:
Thanks for the insights

Reply:
Just throwing in that I don't think direct `column` comparison needs to be supported - Users should be using `Dataframes` and `Series` and not obligated to deal with any deeper support classes.